apps/prairielearn/src/pages/instructorAssessmentQuestions/editQuestionModal.html.ts

```diff
@@ -0,0 +1,184 @@
+import { html } from '@prairielearn/html';
+
+import { Modal } from '../../components/Modal.html.js';
+
+export function EditQuestionModal({
+  newQuestion,
+  question,
+  zoneIndex,
+  questionIndex,
+  alternativeIndex,
+  assessmentType,
+}: {
+  newQuestion: boolean;
+  question?: Record<string, any>;
+  urlPrefix?: string;
+  assessmentInstanceId?: string;
+  zoneIndex: number;
+  questionIndex: number;
+  alternativeIndex?: number;
+  assessmentType: string;
+}) {
+  return Modal({
+    id: 'editQuestionModal',
+    title: newQuestion ? 'Add Question' : 'Update Question',
+    body: html`
+      <input type="hidden" name="zoneIndex" id="zoneIndexInput" value="${zoneIndex}" />
```

Comment:
What are all these hidden inputs for?

Reply:
To pass information back to the client script. Is there a better way to do this?